naming/dns_resolver_pre18.go

```diff
@@ -0,0 +1,16 @@
+// +build go1.6, !go1.8
```

Comment:
By the existing convention, we would name this file "go17.go" (and "go18.go" for the other).